### PR TITLE
Upgrade demo site

### PR DESCRIPTION
**Pull request by timbonicus**
Signed-off-by: Tim Hansen <timbonicush@spotify.com>

### Updates the demo site to include a few more batteries:

- Similar EntityPage/App updates as https://github.com/backstage/backstage/pull/5032
- Adds example systems, users, groups, domains, resources
- Adds search, explore, badges, todo plugins
- Removes register-component, rollbar and lighthouse as they are not demo-site friendly

Obsoletes #23 by doing the same upgrades.

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -42,16 +42,32 @@ auth:
 
 catalog:
```
Comment (freben) [inline app-config.yaml:43]: This should be marked readonly now right?! https://github.com/backstage/backstage/pull/5034

Comment (freben) [reply app-config.yaml:43]: Oh maybe that isn't released...

Comment (benjdlambert) [reply app-config.yaml:43]: Also I think we still might want to throw our own errors for nice error messages in the frontend for the demo site either way?

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 2]
```diff
@@ -13,28 +13,50 @@
...
+  EntityPageLayout,
+  AboutCard,
+  EntityHasComponentsCard,
+  EntityHasSystemsCard,
+} from '@backstage/plugin-catalog';
 import { useEntity } from '@backstage/plugin-catalog-react';
-import { ApiEntity, Entity } from '@backstage/catalog-model';
-import { Grid } from '@material-ui/core';
-import { WarningPanel } from '@backstage/core';
+import {
```
Comment (benjdlambert) [inline packages/app/src/components/catalog/EntityPage.tsx:44]: Not sure that this plugin is worth demoing, We don't have anything building on CircleCI

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 3]
```diff
@@ -13,28 +13,50 @@
...
+  Router as CircleCIRouter,
+  isPluginApplicableToEntity as isCircleCIAvailable,
+} from '@backstage/plugin-circleci';
+import {
+  Router as GitHubActionsRouter,
```
Comment (benjdlambert) [inline packages/app/src/components/catalog/EntityPage.tsx:49]: This plugin needs Github auth to be setup, can't remember if we have that in place, worth a check.

File: app-config.yaml (hunk 1 above)
Comment (timbonicus) [reply app-config.yaml:43]: @benjdlambert Do you mean we should include `catalog-import` for demo purposes but throw errors if you try to create something? The catalog backend does [reject post/delete](https://github.com/backstage/demo/blob/master/packages/backend/src/plugins/catalog.ts#L32), I could test how that manifests on the frontend. I had removed registration for now assuming it wouldn't be viable on the demo.

Comment (benjdlambert) [reply app-config.yaml:43]: I don't think we have the catalog plugin integrated right? I'd prefer to keep it that way as it's not going to do anything anyway. I meant keeping the post/delete rejection too so people can't POST requests to register anything but with nice error messages. @freben is talking about the readonly mode but I'm not sure if the error messages are nice for demo purposes from that functionality so we'd probably want to keep the overrides in the router anyway.

**Pull request merged**